download-release.sh

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env bash
```

Comment:
Random comment: do you want to enable `shellcheck` on this repo prior to this commit?

Reply:
I just ran into this comment randomly, sorry for not replying earlier.

This is actually a good reason to move scripts I originally put into this repo to the main one. I'm planning to do that eventually. I already did that with bytecode PR check.

Not sure about this particular script though. I'd like to salvage it if possible but currently I'm not sure if we actually have a use for it. We should probably talk about it on one of the planning meetings but this currently isn't too high on the priority list for me.

Reply:
Not sure it makes sense moving the `s3-sync.sh` to another repo, so perhaps better to have shellcheck here in any case.

Reply:
Well, I think that even `s3-sync.sh` would fit right in with other scripts that are already in https://github.com/ethereum/solidity/tree/develop/scripts/solc-bin. I even think it might make sense to move `update` there. `solc-bin` with it's current size is not very convenient to use for coding and after doing that for some time I'm more and more convinced that it should contain only binaries and nothing else.

One downside of having code in this repo is that accessing it requires getting a checkout, which is big and wastes a lot of time. This is sometimes an obstacle when you want to have multliple jobs in a single workflow and not all of them need to access binaries. It's not that hard to work around - you can pass the scripts as artifacts or download individual files via github API - but it's still a bit annoying when you need to do this.

To be honest I'm not strongly convinced that we **must** have everything in the main repo - I see both upsides and downsides - I'm just currently leaning more towards that option. Having shellcheck here would not be bad either.

Reply:
Also, it would be nice to have some of these scripts rewritten in Python eventually and for that we already have nice linting and testing infrastructure in main repo. For example the bytecode comparison check for PRs is written in Bash and this is the main reason why I gave up on running it on Windows binaries (they're being ignored currently). Originally I wanted to support them but I realized that I'd have to write a separate PowerShell or CMD version of the scripts just for that one platform and the PR check itself was really just a side task.